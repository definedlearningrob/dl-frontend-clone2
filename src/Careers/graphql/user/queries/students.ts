import { gql } from '@apollo/client';

export default gql`
  query Students($page: Int, $perPage: Int, $filter: StudentFilter, $scope: ArchivableStatus) {
    students(page: $page, perPage: $perPage, filter: $filter, scope: $scope) {
      nodes {
        uuid
        firstName
        lastName
        entity {
          uuid
          name
        }
      }
      nodesCount
      pagesCount
    }
  }
`;
