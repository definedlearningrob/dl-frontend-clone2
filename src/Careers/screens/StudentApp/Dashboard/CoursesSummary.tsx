import { Trans, useTranslation } from 'react-i18next';
import { match } from 'ts-pattern';
import { useToggle } from 'react-use';
import { useDashboardRecentResourcesQuery } from '@graphql/dc/students/hooks';
import { DashboardRecentResourceTypes } from '@graphql/dc/students/types';

import { ReactComponent as ChevronRightIcon } from '@shared/svg/chevron_right.svg';
import { ReactComponent as CourseIcon } from '@shared/svg/book_opened.svg';
import { ReactComponent as LaptopIcon } from '@shared/svg/laptop.svg';
import Modal from '@shared/components/Modal/Modal';
import { GenericCard } from '@shared/components/GenericCard/GenericCard';
import SkeletonRectangle from '@shared/components/Skeleton/Rectangle/Rectangle';
import { cx } from '@shared/utils/cx';
import Button from '@shared/components/Button/Button';

export const CoursesSummary = () => {
  const { t } = useTranslation();

  const [isModalOpen, toggleIsModalOpen] = useToggle(false);

  const { data, loading } = useDashboardRecentResourcesQuery();

  if (loading) {
    return (
      <div className='h-[22px] flex items-center'>
        <SkeletonRectangle className='!w-1/2' size='full-width' />
      </div>
    );
  }

  if (!data) {
    return (
      <p className='mb-0 text-xxs xxxl:text-xs text-neutral-700'>
        {t('student.dashboard.noCourses')}
      </p>
    );
  }

  const coursesCount = data.dashboardRecentResources.length;

  const dashboardCoursesCount = Math.min(coursesCount || 0, 4);

  const translationKey = match(coursesCount)
    .with(0, () => 'student.dashboard.noCourses')
    .when(
      (value) => 1 <= value && value <= 4,
      () => 'student.dashboard.fewCourses'
    )
    .otherwise(() => 'student.dashboard.moreThanFourCourses');

  const modalClassName = cx('!max-h-[85vh]', {
    'h-[480px] xxxl:h-[690px]': coursesCount <= 8,
    'h-[600px] xxxl:h-[800px]': coursesCount > 9,
  });

  return (
    <>
      <p className='mb-0 text-xxs xxxl:text-xs text-neutral-700'>
        <Trans
          components={{
            mediumText: <span className='font-medium' />,
            modalLink: (
              <Button
                children=''
                className='!text-xxs xxxl:!text-xs !inline-block'
                variant='link'
                onClick={toggleIsModalOpen}
              />
            ),
          }}
          i18nKey={translationKey}
          values={{
            coursesCount: dashboardCoursesCount,
            count: coursesCount,
          }}
        />
      </p>
      {isModalOpen && (
        <Modal className={modalClassName} variant='wide' onDismiss={toggleIsModalOpen}>
          <Modal.Header>
            <Modal.Heading>
              <Trans
                components={{
                  neutralText: <span className='text-neutral-600' />,
                }}
                i18nKey='student.dashboard.allActivities'
                values={{
                  count: coursesCount,
                }}
              />
            </Modal.Heading>
          </Modal.Header>
          <Modal.Body className='!max-h-[80vh] !mb-0 !pe-sm'>
            <div className='bg-neutral-200 rounded-sm'>
              <div className='grid grid-cols-3 xxxl:grid-cols-4 gap-sm !h-fit p-sm'>
                {data.dashboardRecentResources.map((resource, index) => {
                  const { name, thumbnailUrl, resourceType, resourceId, pathways, collection } =
                    resource;

                  const collectionName = collection?.name;

                  const isCourse = resourceType === DashboardRecentResourceTypes.COURSE;

                  const typeIconTooltipMessage = isCourse
                    ? t('student.dashboard.course')
                    : t('student.dashboard.virtualInternship');

                  const linkTo = isCourse
                    ? `/courses/${resourceId}`
                    : `/opportunities/${resourceId}`;

                  const TypeIcon = isCourse ? CourseIcon : LaptopIcon;

                  return (
                    <div key={index} className='h-[168px] xxxl:h-[268px]'>
                      <GenericCard
                        Icon={ChevronRightIcon}
                        TypeIcon={TypeIcon}
                        backgroundUrl={thumbnailUrl}
                        pathways={pathways}
                        subTitle={collectionName}
                        title={name}
                        to={linkTo}
                        typeIconTooltipMessage={typeIconTooltipMessage}
                      />
                    </div>
                  );
                })}
              </div>
            </div>
          </Modal.Body>
        </Modal>
      )}
    </>
  );
};
