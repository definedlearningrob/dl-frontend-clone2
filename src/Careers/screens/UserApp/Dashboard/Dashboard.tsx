import { Redirect } from 'react-router';

import EntityAdminView from '@dc/components/User/Dashboard/EntityAdminView/EntityAdminView';
import SharedMainContent from '@dc/shared/MainContent/MainContent';
import SystemAdminView from '@dc/components/User/Dashboard/SystemAdminView/SystemAdminView';
import TeacherView from '@dc/components/User/Dashboard/TeacherView/TeacherView';
import useUserInfo from '@dc/hooks/useUserInfo';
import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import { Roles } from '@dc/resources/enums';

function UserAppDashboard() {
  const { userInfo } = useUserInfo<TUserInfo>();

  if (!userInfo?.role) {
    return <Redirect to='/' />;
  }

  const renderCorrectDashboard = () =>
    ({
      [Roles.SALES_ADMIN]: <SystemAdminView />,
      [Roles.SYSTEM_ADMIN]: <SystemAdminView />,
      [Roles.ENTITY_ADMIN]: <EntityAdminView />,
      [Roles.TEACHER]: <TeacherView />,
    }[userInfo.role]);

  return (
    <SharedMainContent>
      <div className='user-app-dashboard'>{renderCorrectDashboard()}</div>
    </SharedMainContent>
  );
}

export default UserAppDashboard;
