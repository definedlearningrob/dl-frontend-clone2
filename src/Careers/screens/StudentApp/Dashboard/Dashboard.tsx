import { Trans } from 'react-i18next';
import { useRef } from 'react';
import { useCustomCatalogOverviewQuery } from '@graphql/dc/shared/hooks';
import { isEmpty } from 'lodash-es';

import useUserInfo from '@dc/hooks/useUserInfo';
import { TStudentInfo } from '@dc/graphql/student/queries/userInfo';
import { CoursesSummary } from '@dc/screens/StudentApp/Dashboard/CoursesSummary';
import { LatestActivitySection } from '@dc/screens/StudentApp/Dashboard/LatestActivitySection/LatestActivitySection';
import { OnboardingFirstLogin } from '@dc/components/Onboarding/FirstLogin/FirstLogin';
import { QuickLinks } from '@dc/screens/StudentApp/Dashboard/QuickLinks/QuickLinks';
import { ModuleLinks } from '@dc/screens/StudentApp/Dashboard/ModuleLinks/ModuleLinks';

import { WelcomeMessage } from '@shared/components/WelcomeMessage';
import { useResizeWidthObserver } from '@shared/hooks/useResizeWidthObserver';
import { MainContent } from '@shared/components/MainContent/MainContent';
import Card from '@shared/components/Card/Card';

import { CustomCatalogSection } from './CustomCatalogSection/CustomCatalogSection';
import { PlanProgressCard } from './PlanProgressCard/PlanProgressCard';

const ADDITIONAL_OFFSET = 8;

type Props = {
  beforeOnboarding: boolean;
};

export const Dashboard = ({ beforeOnboarding }: Props) => {
  const { data: customCatalogData } = useCustomCatalogOverviewQuery();
  const { userInfo } = useUserInfo<TStudentInfo>();

  const contentWrapperRef = useRef(null);
  const nameRef = useRef(null);

  const nameWidth = useResizeWidthObserver(nameRef);
  const welcomeMessageWidth = useResizeWidthObserver(contentWrapperRef);

  const {
    lastName,
    firstName,
    welcomeMessage,
    hasPlans,
    hasOpportunitiesEnabled,
    postSecondaryApplicationsEnabled,
  } = userInfo;

  const hasCustomCatalog = !isEmpty(customCatalogData?.careersCatalog?.id);

  return (
    <MainContent className='h-full xl:h-[theme(layout.containerHeight)] xxxl:!pt-sm !pt-xs relative'>
      {beforeOnboarding && <OnboardingFirstLogin />}
      <div className='h-full flex flex-col xl:flex-row gap-base xxxl:gap-md'>
        <div className='flex flex-col grow gap-sm xxxl:gap-md'>
          <div>
            <div className='flex items-start gap-xs'>
              <h4 ref={nameRef} className='text-base xxxl:text-lg !mb-xs xxxl:!mb-sm'>
                <Trans
                  components={{
                    neutralText: <span className='text-neutral-600' />,
                  }}
                  i18nKey='student.dashboard.helloMessage'
                  values={{
                    firstName,
                    lastName,
                  }}
                />
              </h4>
              {welcomeMessage && (
                <WelcomeMessage
                  leftOffset={-nameWidth - ADDITIONAL_OFFSET}
                  welcomeMessage={welcomeMessage}
                  width={welcomeMessageWidth}
                />
              )}
            </div>
            <CoursesSummary />
          </div>
          <div className='flex flex-col gap-base min-h-0 grow'>
            <div
              ref={contentWrapperRef}
              className='flex gap-sm shrink min-h-0 xxxl:gap-base grow basis-[349px] xxxl:basis-[560px]'>
              <div className='grow  min-h-0'>
                <LatestActivitySection />
              </div>
              {hasCustomCatalog && (
                <div className='shrink-0 basis-[240px] xxxl:basis-[380px]'>
                  <CustomCatalogSection />
                </div>
              )}
            </div>
            <div className='grow lg:h-[205px] xxxl:h-[296px] h-full'>
              <ModuleLinks
                hasOpportunitiesEnabled={hasOpportunitiesEnabled}
                hasPostSecondaryApplicationsEnabled={postSecondaryApplicationsEnabled}
              />
            </div>
          </div>
        </div>

        <div className='shrink-0 basis-[318px] xxxl:basis-[512px] h-full flex flex-row xl:flex-col gap-base xxxl:gap-md'>
          {hasPlans && <PlanProgressCard />}
          <Card className='grow'>
            <QuickLinks />
          </Card>
        </div>
      </div>
    </MainContent>
  );
};
