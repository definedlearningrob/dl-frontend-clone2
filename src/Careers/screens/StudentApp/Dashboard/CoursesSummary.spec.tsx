import { MockedProvider } from '@apollo/client/testing';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { DashboardRecentResourcesDocument } from '@graphql/dc/students/hooks';

import { renderWithRouter } from '@dc/utils/test';
import { CoursesSummary } from '@dc/screens/StudentApp/Dashboard/CoursesSummary';

const activitiesSpy = jest.fn();

const getActivitiesMock = (count: number) => {
  const dashboardRecentResources = Array.from({ length: count }, (_, index) => ({
    resourceId: (index + 1).toString(),
    resourceType: 'VIRTUAL_INTERNSHIP',
    name: 'Test',
    pathways: [
      {
        name: 'Agribusiness Systems',
      },
    ],
    thumbnailUrl: 'http://www.test.com/test.jpg',
    imageUrl: 'http://www.test.com/test.jpg',
    collection: null,
    updatedAt: '2024-03-04T14:55:28Z',
  }));

  return {
    request: {
      query: DashboardRecentResourcesDocument,
      variables: {},
    },
    result: () => {
      activitiesSpy();

      return {
        data: {
          dashboardRecentResources,
        },
      };
    },
  };
};

describe('CoursesSummary', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const testCases = [
    [0, 'Here you can access your most recent activities, you can add more through courses.'],
    [1, 'Here you can access your 1 most recent activity.'],
    [2, 'Here you can access your 2 most recent activities.'],
    [3, 'Here you can access your 3 most recent activities.'],
    [4, 'Here you can access your 4 most recent activities.'],
    [
      5,
      'Here you can access your 4 most recent activities from 5 total. You can add more through courses.',
    ],
    [
      15,
      'Here you can access your 4 most recent activities from 15 total. You can add more through courses.',
    ],
  ];

  it.each(testCases)(
    'should render correct summary for %s activities',
    async (activitiesCount, expectedSummary) => {
      const { container } = renderWithRouter(
        <MockedProvider mocks={[getActivitiesMock(Number(activitiesCount))]}>
          <CoursesSummary />
        </MockedProvider>
      );

      await waitFor(() => expect(activitiesSpy).toHaveBeenCalled());

      expect(container.textContent).toEqual(expectedSummary);

      expect(container).toMatchSnapshot();
    }
  );

  it('should render modal', async () => {
    renderWithRouter(
      <MockedProvider mocks={[getActivitiesMock(5)]}>
        <CoursesSummary />
      </MockedProvider>
    );

    await waitFor(() => expect(activitiesSpy).toHaveBeenCalled());

    const modalToggle = screen.getByText(/5 total/i);

    userEvent.click(modalToggle);

    const modal = await screen.findByRole('dialog');
    const modalHeader = await screen.getByText(/All Activities/i);

    expect(modal).toBeInTheDocument();
    expect(modalHeader.textContent).toEqual('All Activities (5)');

    expect(modal).toMatchSnapshot();
  });
});
