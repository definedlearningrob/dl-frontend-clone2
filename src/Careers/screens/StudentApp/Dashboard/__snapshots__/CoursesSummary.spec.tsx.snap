// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesSummary should render correct summary for 0 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your most recent activities, you can add more through courses.
  </p>
</div>
`;

exports[`CoursesSummary should render correct summary for 1 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your 
    <span
      class="font-medium"
    >
      1 most recent
    </span>
     activity.
  </p>
</div>
`;

exports[`CoursesSummary should render correct summary for 2 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your 
    <span
      class="font-medium"
    >
      2 most recent
    </span>
     activities.
  </p>
</div>
`;

exports[`CoursesSummary should render correct summary for 3 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your 
    <span
      class="font-medium"
    >
      3 most recent
    </span>
     activities.
  </p>
</div>
`;

exports[`CoursesSummary should render correct summary for 4 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your 
    <span
      class="font-medium"
    >
      4 most recent
    </span>
     activities.
  </p>
</div>
`;

exports[`CoursesSummary should render correct summary for 5 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your 
    <span
      class="font-medium"
    >
      4 most recent
    </span>
     activities from 
    <button
      class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] !text-xxs xxxl:!text-xs !inline-block -link py-xs px-base leading-[22px] text-white"
      data-testid="button"
      type="button"
    >
      <span
        class="items-center flex justify-center gap-xs"
      >
        5 total.
      </span>
    </button>
     You can add more through courses.
  </p>
</div>
`;

exports[`CoursesSummary should render correct summary for 15 activities 1`] = `
<div>
  <p
    class="mb-0 text-xxs xxxl:text-xs text-neutral-700"
  >
    Here you can access your 
    <span
      class="font-medium"
    >
      4 most recent
    </span>
     activities from 
    <button
      class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] !text-xxs xxxl:!text-xs !inline-block -link py-xs px-base leading-[22px] text-white"
      data-testid="button"
      type="button"
    >
      <span
        class="items-center flex justify-center gap-xs"
      >
        15 total.
      </span>
    </button>
     You can add more through courses.
  </p>
</div>
`;

exports[`CoursesSummary should render modal 1`] = `
<div
  aria-label="Modal"
  aria-modal="true"
  class="dialogContent modal bg-white flex flex-col shadow-300 pb-base xxxl:pb-md overflow-hidden my-[40px] xxxl:my-[80px] transition-[width] duration-500 ease-in-out rounded-sm outline-none w-[800px] w-[920px] xxxl:w-[1200px] !max-h-[85vh] h-[480px] xxxl:h-[690px]"
  data-reach-dialog-content=""
  data-state="open"
  role="dialog"
  tabindex="-1"
>
  <div
    class="modal__header flex flex-row justify-between items-start px-base xxxl:px-md pt-base relative text-neutral-800 xxxl:pt-md xxxl:pb-0 mb-sm"
    data-testid="modal-header"
  >
    <h4
      class="modal__heading mb-0 text-lg"
      data-testid="modal-heading"
    >
      All Activities 
      <span
        class="text-neutral-600"
      >
        (5)
      </span>
    </h4>
    <button
      aria-label="Modal close"
      class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 w-md h-md"
      data-testid="modal-close-button"
      type="button"
    >
      <div
        class="flex"
      >
        <div
          class="flex align-center justify-center grow-0 shrink-0"
          data-testid="icon"
        >
          <div
            class="h-base w-base"
          />
        </div>
      </div>
    </button>
  </div>
  <div
    class="modal__body grow text-font-primary leading-base scrollbar overflow-y-auto overflow-x-hidden py-0 px-base xxxl:px-md mb-sm xxxl:mb-base max-h-[60vh] !max-h-[80vh] !mb-0 !pe-sm"
    data-testid="modal-body"
  >
    <div
      class="bg-neutral-200 rounded-sm"
    >
      <div
        class="grid grid-cols-3 xxxl:grid-cols-4 gap-sm !h-fit p-sm"
      >
        <div
          class="h-[168px] xxxl:h-[268px]"
        >
          <a
            href="/opportunities/1"
          >
            <div
              aria-label="Test"
              class="cursor-pointer rounded-sm h-full relative overflow-hidden whitespace-normal before:absolute before:!bg-generic-card-gradient before:inset-0 hover:before:!bg-generic-card-gradient-hovered hover:before:bg-[#3c42583d] hover:shadow-300 group/card"
              role="none"
            >
              <img
                class="w-full h-full object-cover"
                src="http://www.test.com/test.jpg"
              />
              <div
                class="absolute inset-0 group-hover/card:pb-sm group-hover/card:xxxl:pb-base px-sm pt-sm pb-base xxxl:px-base xxxl:pt-base xxxl:pb-md"
              >
                <div
                  class="relative h-full flex flex-col justify-end"
                >
                  <div>
                    <div
                      class="flex flex-row-reverse justify-between absolute top-0 right-0 left-0 items-center"
                    >
                      <div
                        class="border border-white rounded-full overflow-hidden"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 transition-colors ms-auto h-md w-md xxxl:h-lg xxxl:w-lg text-white group-hover/card:bg-white group-hover/card:text-primary-500 p-xxs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="w-fit"
                        />
                      </div>
                    </div>
                  </div>
                  <span
                    data-state="closed"
                  >
                    <div
                      class="flex text-neutral-300 items-center gap-xxs mb-xxxs"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <div
                        class="text-xxs xxxl:text-xs italic leading-lg truncate"
                      >
                        Agribusiness Systems
                      </div>
                    </div>
                  </span>
                  <span
                    data-state="closed"
                  >
                    <span
                      data-testid="text-highlighter"
                    >
                      <h5
                        class="text-white text-xs xxxl:text-lg leading-[1.2] line-clamp-3 mb-0 highlightible"
                      >
                        Test
                      </h5>
                    </span>
                  </span>
                  <div
                    class="w-full overflow-hidden"
                    style="height: 0px; transform: translateY(16px) translateZ(0);"
                  >
                    <div
                      class="pt-xxs flex items-center gap-xs"
                    >
                      <div
                        class="ms-auto"
                      >
                        <span
                          class="min-w-0"
                          data-state="closed"
                        >
                          <div
                            class="flex"
                          >
                            <div
                              class="flex align-center justify-center grow-0 shrink-0 text-white"
                              data-testid="icon"
                            >
                              <div
                                class="h-base w-base"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="h-[168px] xxxl:h-[268px]"
        >
          <a
            href="/opportunities/2"
          >
            <div
              aria-label="Test"
              class="cursor-pointer rounded-sm h-full relative overflow-hidden whitespace-normal before:absolute before:!bg-generic-card-gradient before:inset-0 hover:before:!bg-generic-card-gradient-hovered hover:before:bg-[#3c42583d] hover:shadow-300 group/card"
              role="none"
            >
              <img
                class="w-full h-full object-cover"
                src="http://www.test.com/test.jpg"
              />
              <div
                class="absolute inset-0 group-hover/card:pb-sm group-hover/card:xxxl:pb-base px-sm pt-sm pb-base xxxl:px-base xxxl:pt-base xxxl:pb-md"
              >
                <div
                  class="relative h-full flex flex-col justify-end"
                >
                  <div>
                    <div
                      class="flex flex-row-reverse justify-between absolute top-0 right-0 left-0 items-center"
                    >
                      <div
                        class="border border-white rounded-full overflow-hidden"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 transition-colors ms-auto h-md w-md xxxl:h-lg xxxl:w-lg text-white group-hover/card:bg-white group-hover/card:text-primary-500 p-xxs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="w-fit"
                        />
                      </div>
                    </div>
                  </div>
                  <span
                    data-state="closed"
                  >
                    <div
                      class="flex text-neutral-300 items-center gap-xxs mb-xxxs"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <div
                        class="text-xxs xxxl:text-xs italic leading-lg truncate"
                      >
                        Agribusiness Systems
                      </div>
                    </div>
                  </span>
                  <span
                    data-state="closed"
                  >
                    <span
                      data-testid="text-highlighter"
                    >
                      <h5
                        class="text-white text-xs xxxl:text-lg leading-[1.2] line-clamp-3 mb-0 highlightible"
                      >
                        Test
                      </h5>
                    </span>
                  </span>
                  <div
                    class="w-full overflow-hidden"
                    style="height: 0px; transform: translateY(16px) translateZ(0);"
                  >
                    <div
                      class="pt-xxs flex items-center gap-xs"
                    >
                      <div
                        class="ms-auto"
                      >
                        <span
                          class="min-w-0"
                          data-state="closed"
                        >
                          <div
                            class="flex"
                          >
                            <div
                              class="flex align-center justify-center grow-0 shrink-0 text-white"
                              data-testid="icon"
                            >
                              <div
                                class="h-base w-base"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="h-[168px] xxxl:h-[268px]"
        >
          <a
            href="/opportunities/3"
          >
            <div
              aria-label="Test"
              class="cursor-pointer rounded-sm h-full relative overflow-hidden whitespace-normal before:absolute before:!bg-generic-card-gradient before:inset-0 hover:before:!bg-generic-card-gradient-hovered hover:before:bg-[#3c42583d] hover:shadow-300 group/card"
              role="none"
            >
              <img
                class="w-full h-full object-cover"
                src="http://www.test.com/test.jpg"
              />
              <div
                class="absolute inset-0 group-hover/card:pb-sm group-hover/card:xxxl:pb-base px-sm pt-sm pb-base xxxl:px-base xxxl:pt-base xxxl:pb-md"
              >
                <div
                  class="relative h-full flex flex-col justify-end"
                >
                  <div>
                    <div
                      class="flex flex-row-reverse justify-between absolute top-0 right-0 left-0 items-center"
                    >
                      <div
                        class="border border-white rounded-full overflow-hidden"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 transition-colors ms-auto h-md w-md xxxl:h-lg xxxl:w-lg text-white group-hover/card:bg-white group-hover/card:text-primary-500 p-xxs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="w-fit"
                        />
                      </div>
                    </div>
                  </div>
                  <span
                    data-state="closed"
                  >
                    <div
                      class="flex text-neutral-300 items-center gap-xxs mb-xxxs"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <div
                        class="text-xxs xxxl:text-xs italic leading-lg truncate"
                      >
                        Agribusiness Systems
                      </div>
                    </div>
                  </span>
                  <span
                    data-state="closed"
                  >
                    <span
                      data-testid="text-highlighter"
                    >
                      <h5
                        class="text-white text-xs xxxl:text-lg leading-[1.2] line-clamp-3 mb-0 highlightible"
                      >
                        Test
                      </h5>
                    </span>
                  </span>
                  <div
                    class="w-full overflow-hidden"
                    style="height: 0px; transform: translateY(16px) translateZ(0);"
                  >
                    <div
                      class="pt-xxs flex items-center gap-xs"
                    >
                      <div
                        class="ms-auto"
                      >
                        <span
                          class="min-w-0"
                          data-state="closed"
                        >
                          <div
                            class="flex"
                          >
                            <div
                              class="flex align-center justify-center grow-0 shrink-0 text-white"
                              data-testid="icon"
                            >
                              <div
                                class="h-base w-base"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="h-[168px] xxxl:h-[268px]"
        >
          <a
            href="/opportunities/4"
          >
            <div
              aria-label="Test"
              class="cursor-pointer rounded-sm h-full relative overflow-hidden whitespace-normal before:absolute before:!bg-generic-card-gradient before:inset-0 hover:before:!bg-generic-card-gradient-hovered hover:before:bg-[#3c42583d] hover:shadow-300 group/card"
              role="none"
            >
              <img
                class="w-full h-full object-cover"
                src="http://www.test.com/test.jpg"
              />
              <div
                class="absolute inset-0 group-hover/card:pb-sm group-hover/card:xxxl:pb-base px-sm pt-sm pb-base xxxl:px-base xxxl:pt-base xxxl:pb-md"
              >
                <div
                  class="relative h-full flex flex-col justify-end"
                >
                  <div>
                    <div
                      class="flex flex-row-reverse justify-between absolute top-0 right-0 left-0 items-center"
                    >
                      <div
                        class="border border-white rounded-full overflow-hidden"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 transition-colors ms-auto h-md w-md xxxl:h-lg xxxl:w-lg text-white group-hover/card:bg-white group-hover/card:text-primary-500 p-xxs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="w-fit"
                        />
                      </div>
                    </div>
                  </div>
                  <span
                    data-state="closed"
                  >
                    <div
                      class="flex text-neutral-300 items-center gap-xxs mb-xxxs"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <div
                        class="text-xxs xxxl:text-xs italic leading-lg truncate"
                      >
                        Agribusiness Systems
                      </div>
                    </div>
                  </span>
                  <span
                    data-state="closed"
                  >
                    <span
                      data-testid="text-highlighter"
                    >
                      <h5
                        class="text-white text-xs xxxl:text-lg leading-[1.2] line-clamp-3 mb-0 highlightible"
                      >
                        Test
                      </h5>
                    </span>
                  </span>
                  <div
                    class="w-full overflow-hidden"
                    style="height: 0px; transform: translateY(16px) translateZ(0);"
                  >
                    <div
                      class="pt-xxs flex items-center gap-xs"
                    >
                      <div
                        class="ms-auto"
                      >
                        <span
                          class="min-w-0"
                          data-state="closed"
                        >
                          <div
                            class="flex"
                          >
                            <div
                              class="flex align-center justify-center grow-0 shrink-0 text-white"
                              data-testid="icon"
                            >
                              <div
                                class="h-base w-base"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="h-[168px] xxxl:h-[268px]"
        >
          <a
            href="/opportunities/5"
          >
            <div
              aria-label="Test"
              class="cursor-pointer rounded-sm h-full relative overflow-hidden whitespace-normal before:absolute before:!bg-generic-card-gradient before:inset-0 hover:before:!bg-generic-card-gradient-hovered hover:before:bg-[#3c42583d] hover:shadow-300 group/card"
              role="none"
            >
              <img
                class="w-full h-full object-cover"
                src="http://www.test.com/test.jpg"
              />
              <div
                class="absolute inset-0 group-hover/card:pb-sm group-hover/card:xxxl:pb-base px-sm pt-sm pb-base xxxl:px-base xxxl:pt-base xxxl:pb-md"
              >
                <div
                  class="relative h-full flex flex-col justify-end"
                >
                  <div>
                    <div
                      class="flex flex-row-reverse justify-between absolute top-0 right-0 left-0 items-center"
                    >
                      <div
                        class="border border-white rounded-full overflow-hidden"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 transition-colors ms-auto h-md w-md xxxl:h-lg xxxl:w-lg text-white group-hover/card:bg-white group-hover/card:text-primary-500 p-xxs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="w-fit"
                        />
                      </div>
                    </div>
                  </div>
                  <span
                    data-state="closed"
                  >
                    <div
                      class="flex text-neutral-300 items-center gap-xxs mb-xxxs"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <div
                        class="text-xxs xxxl:text-xs italic leading-lg truncate"
                      >
                        Agribusiness Systems
                      </div>
                    </div>
                  </span>
                  <span
                    data-state="closed"
                  >
                    <span
                      data-testid="text-highlighter"
                    >
                      <h5
                        class="text-white text-xs xxxl:text-lg leading-[1.2] line-clamp-3 mb-0 highlightible"
                      >
                        Test
                      </h5>
                    </span>
                  </span>
                  <div
                    class="w-full overflow-hidden"
                    style="height: 0px; transform: translateY(16px) translateZ(0);"
                  >
                    <div
                      class="pt-xxs flex items-center gap-xs"
                    >
                      <div
                        class="ms-auto"
                      >
                        <span
                          class="min-w-0"
                          data-state="closed"
                        >
                          <div
                            class="flex"
                          >
                            <div
                              class="flex align-center justify-center grow-0 shrink-0 text-white"
                              data-testid="icon"
                            >
                              <div
                                class="h-base w-base"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
